File: bank-service/src/main/java/rs/edu/raf/model/card/Card.java
package rs.edu.raf.model.card;

import jakarta.persistence.*;
import lombok.*;
import rs.edu.raf.model.accounts.BankAccount;

import java.math.BigDecimal;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity
public class Card {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String number;
    @Enumerated(value = EnumType.STRING)
    private CardType type;
    @ManyToOne
    private CardIssuer cardIssuer;
    private String name;
    private Long creationDate;
    private Long expirationDate;
    @ManyToOne
    private BankAccount bankAccount;
    private int cvv;
    private BigDecimal cardLimit;
    @Enumerated(value = EnumType.STRING)
    private CardStatus cardStatus;
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/model/card/Card.kt
package org.example.bankservicekotlin.model.card

import jakarta.persistence.*
import org.example.bankservicekotlin.model.accounts.BankAccount
import java.math.BigDecimal

@Entity
class Card(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long?,
    var number: String,
    @Enumerated(value = EnumType.STRING)
    val type: CardType,
    @ManyToOne
    val cardIssuer: CardIssuer,
    val name: String,
    val creationDate: Long,
    val expirationDate: Long,
    @ManyToOne
    val bankAccount: BankAccount,
    val cvv: Int,
    val cardLimit: BigDecimal,
    @Enumerated(value = EnumType.STRING)
    var cardStatus: CardStatus
) {
}